### PR TITLE
feat: improve enrolment flow

### PR DESCRIPTION
**Pull request by hejkerooo**

### CODE REVIEW

File: apps/dsb-client-gateway-api/src/app/modules/enrolment/service/claim-listener.service.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+      return;
+    }
+
+    this.eventEmitter = new EventEmitter();
+    this.rolesToListen = roles;
+
+    const eventsUrl = this.configService.get<string>(
+      'EVENT_SERVER_URL',
+      'identityevents-dev.energyweb.org'
```
Comment (dawidgil) [inline apps/dsb-client-gateway-api/src/app/modules/enrolment/service/claim-listener.service.ts:42]: Isn't it better to have exported const for such values?

File: apps/dsb-client-gateway-api/src/app/modules/enrolment/service/role-listener.service.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      this.logger.error('No balance');
+
+      throw new NotEnoughBalanceException(identity.address);
+    }
+
+    const enrolment: Enrolment =
+      await this.enrolmentService.generateEnrolment();
+
+    /**
+     * TODO:
```
Comment (dawidgil) [inline apps/dsb-client-gateway-api/src/app/modules/enrolment/service/role-listener.service.ts:60]: comments in polish :D

File: libs/dsb-client-gateway-iam-client/src/lib/service/iam.service.ts [hunk 3]
```diff
@@ -30,11 +31,44 @@ export class IamService {
...
+        return RoleStatus.AWAITING_APPROVAL;
+      }
+
+      return RoleStatus.NOT_ENROLLED;
+    };
+
+    return claims.map((claim) => {
+      return {
+        namespace: claim.claimType,
+        status: getClaimStatus(claim),
```
Comment (dawidgil) [inline libs/dsb-client-gateway-iam-client/src/lib/service/iam.service.ts:57]: getClaimStatus do not return `Synced` status.

File: package.json [hunk 4]
```diff
@@ -48,7 +49,7 @@
     "form-data": "^4.0.0",
     "formidable": "^2.0.1",
     "hdkey": "^2.0.1",
-    "iam-client-lib": "^4.0.1",
+    "iam-client-lib": "^4.2.0",
```
Comment (dawidgil) [inline package.json:52]: fyi, there is version 5 of ICL. There were no breaking changes afaik between 4.2 and 5, and there are some fixes to role enrolment.

File: apps/dsb-client-gateway-api/src/app/modules/enrolment/service/claim-listener.service.ts (hunk 1 above)
Comment (hejkerooo) [reply apps/dsb-client-gateway-api/src/app/modules/enrolment/service/claim-listener.service.ts:42]: actually all those values have defaults defined in `EnvironmentVariables` so I can remove it

File: libs/dsb-client-gateway-iam-client/src/lib/service/iam.service.ts (hunk 3 above)
Comment (hejkerooo) [reply libs/dsb-client-gateway-iam-client/src/lib/service/iam.service.ts:57]: this flow is different than enrolment flow

**Review by dawidgil — APPROVED** [review]

**Pull request merged**